debugger/src/debugger/debug_entry.rs

```diff
@@ -339,631 +320,584 @@ impl Debugger {
         // Handling the initialize, and Attach/Launch requests here in this method,
         // before entering the iterative loop that processes requests through the process_request method.
 
-        // Initialize request.
-        let initialize_request = loop {
-            let current_request = if let Some(request) = debug_adapter.listen_for_request()? {
-                request
-            } else {
-                continue;
-            };
+        // Initialize request
+        self.handle_initialize(&mut debug_adapter)?;
 
-            match current_request.command.as_str() {
-                "initialize" => break current_request, // We have lift off.
-                other => {
-                    let command = other.to_string();
-
-                    debug_adapter.send_response::<()>(
-                        current_request,
-                        Err(
-                            anyhow!("Initial command was '{}', expected 'initialize'", command)
-                                .into(),
-                        ),
-                    )?;
-                    return Err(DebuggerError::Other(anyhow!(
-                        "Initial command was '{}', expected 'initialize'",
-                        command
-                    )));
-                }
-            };
-        };
+        // Process either the Launch or Attach request.
+        let (mut debug_adapter, mut session_data) = self.handle_launch_attach(debug_adapter)?;
```

Comment:
Somehow during an "attach" request, we no longer send back a response to the request (this used to happen on line 874 in master branch). The result is that the the "attach" request never completes, and according to the DAP Client, the debug session is never initialized.